### PR TITLE
JAVA-1438: QueryBuilder check for empty orderings

### PR DESCRIPTION
**Pull request by podnov**
Added orderings.isEmpty check before QueryBuilder adds ORDER BY clause

### CODE REVIEW

**Comment by tolbertam** [comment]
This looks good to me, thanks for your contribution @podnov!   I changed the target branch to 3.2.x as I think this is straightforward enough to fix in a patch release.

**Comment by tolbertam** [comment]
A side note for whoever merges this that we should add a changelog entry for this (I am forgoing doing this at the moment as it would inevitably create a merge conflict, so it makes sense to wait until we merge).

**Review by tolbertam — APPROVED** [review]

File: driver-core/src/main/java/com/datastax/driver/core/querybuilder/Select.java
Comment (newkek) [inline driver-core/src/main/java/com/datastax/driver/core/querybuilder/Select.java]: I don't think we should silently ignore if the user calls QueryBuilder methods that are not valid to create a CQL query. If the argument in the `orderBy` method is not correct for a CQL query we should throw an exception. As it is done for other query builder methods, like `limit()` if the argument is not valid for Cassandra (limit < 0), we should not ignore it when creating the query

**Review by newkek — CHANGES_REQUESTED** [review]

**Comment by newkek** [comment]
@podnov Thank you very much for the contribution! I mentioned a small change in the review, and we are hoping to get this merged soon, so I'll go ahead and do that change (will keep the commit attributed to you) and if that change does not suit your needs, please re-open the JIRA, thanks!

**Review by newkek — APPROVED** [review]

File: driver-core/src/main/java/com/datastax/driver/core/querybuilder/Select.java
Comment (podnov) [reply driver-core/src/main/java/com/datastax/driver/core/querybuilder/Select.java]: You've certainly fixed the issue as reported in JIRA, thanks for that.  My only feedback would be that when the person writing code against the querybuilder doesn't know exactly what the end user is going to give them as a query, not accepting an empty list of orderings can make for some ugly builder code:

https://gist.github.com/podnov/b67f7e57cf702df579b68b67dde839c7

I'm sure there are other ways to write this, but the ones immediately apparent to me don't seem terribly straight forward.

**Pull request merged**
